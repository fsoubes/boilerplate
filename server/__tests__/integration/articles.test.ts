import { gql } from "apollo-server-express";
import testServer from "../utils/testServer";
import mongoose from "mongoose";
global.fetch = require("node-fetch");

const ArticleFragment = gql`
  {
    title
    description
    image_url
    tags
    source
    social
    article
    isPublished
    author {
      nickname
      email
    }
  }
`;

const ADD_ARTICLE = gql`
  mutation AddArticle(
    $title: String!
    $description: String
    $image_url: String
    $tags: [String!]
    $source: [String!]
    $social: [String!]
    $article: String
    $isPublished: Boolean
  ) {
    addArticle(
      blog: {
        title: $title
        description: $description
        image_url: $image_url
        tags: $tags
        source: $source
        social: $social
        article: $article
        isPublished: $isPublished
      }
    ) {
        ...${ArticleFragment}
    }
  }
`;

const GET_ARTICLE = gql`
  query getSingleArticle($articleId: ObjectId!) {
    getSingleArticle(articleId: $articleId) {
      ...${ArticleFragment}
    }
  }
`;

const GET_ARTICLES = gql`
  query GetArticles($limit: Float!, $cursor: String) {
    getArticles(limit: $limit, cursor: $cursor) {
      edges {
        ...${ArticleFragment}
      }
      pageInfo {
        hasNextPage
        endCursor
      }
    }
  }
`;

describe("Add Article and request it", () => {
  test("Add and Get Article", async () => {
    // ACT
    // use the test server to create a query function

    await testServer("");

    const defaultUser = await mongoose.connection.models.User.findOne({
      email: "bob@bob.fr",
    });

    console.log("Add Mock User with cookie");
    const { query, mutate } = await testServer(defaultUser._id.toString());

    const { data } = await mutate({
      mutation: ADD_ARTICLE,
      variables: {
        title: "NextJs",
        description: "Nextjs is good",
        image_url: "http://dummyimage.com/189x221.bmp/dddddd/000000",
        article: "Mixed hearing loss, unilateral with Apollo and Graphql",
        isPublished: true,
        author: defaultUser._id,
        tags: ["Apollo", "NextJs", "GraphQL"],
        source: ["http://trustme.com", "http://someonetoldme.com"],
        social: ["http://facebook.com", "http://twitter.com"],
      },
    });

    // ASSERT
    // a notification created from you for yourself is not needed. expect a blank array if you are liking your post
    console.log("Test for adding an article");
    expect(data).toMatchSnapshot();

    const lastArticle = await mongoose.connection.models.Blog.findOne({
      title: "NextJs",
    });

    // fetch series
    const getArticle = await query({
      query: GET_ARTICLE,
      variables: { articleId: lastArticle._id },
    });

    // ASSERT
    // a notification created from you for yourself is not needed. expect a blank array if you are liking your post
    console.log("Test for fetching the new article");
    expect(getArticle).toMatchSnapshot();
  });
});

describe("Test pagination for articles", () => {
  test("Pagination Article", async () => {
    // ACT
    // use the test server to create a query function
    const { query } = await testServer("");

    // fetch series
    const { data } = await query({
      query: GET_ARTICLES,
      variables: {
        limit: 46,
        cursor: null,
      },
    });

    // ASSERT
    // a notification created from you for yourself is not needed. expect a blank array if you are liking your post
    console.log("Test pagination");
    expect(data.getArticles).toMatchSnapshot();

    // fetch series
    const getArticleswCursor = await query({
      query: GET_ARTICLES,
      variables: {
        limit: 5,
        cursor: data.getArticles.pageInfo.endCursor,
      },
    });

    // ASSERT
    // a notification created from you for yourself is not needed. expect a blank array if you are liking your post
    console.log("Test pagination");
    expect(getArticleswCursor).toMatchSnapshot();
  });
});
