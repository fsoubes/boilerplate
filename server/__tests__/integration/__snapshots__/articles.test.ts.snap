// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Article and request it Add and Get Article 1`] = `
Object {
  "addArticle": Object {
    "article": "Mixed hearing loss, unilateral with Apollo and Graphql",
    "author": Object {
      "email": "bob@bob.fr",
      "nickname": "bob",
    },
    "description": "Nextjs is good",
    "image_url": "http://dummyimage.com/189x221.bmp/dddddd/000000",
    "isPublished": true,
    "social": Array [
      "http://facebook.com",
      "http://twitter.com",
    ],
    "source": Array [
      "http://trustme.com",
      "http://someonetoldme.com",
    ],
    "tags": Array [
      "Apollo",
      "NextJs",
      "GraphQL",
    ],
    "title": "NextJs",
  },
}
`;

exports[`Add Article and request it Add and Get Article 2`] = `
Object {
  "data": Object {
    "getSingleArticle": Object {
      "article": "Mixed hearing loss, unilateral with Apollo and Graphql",
      "author": Object {
        "email": "bob@bob.fr",
        "nickname": "bob",
      },
      "description": "Nextjs is good",
      "image_url": "http://dummyimage.com/189x221.bmp/dddddd/000000",
      "isPublished": true,
      "social": Array [
        "http://facebook.com",
        "http://twitter.com",
      ],
      "source": Array [
        "http://trustme.com",
        "http://someonetoldme.com",
      ],
      "tags": Array [
        "Apollo",
        "NextJs",
        "GraphQL",
      ],
      "title": "NextJs",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`Test pagination for articles Pagination Article 1`] = `
Object {
  "edges": Array [
    Object {
      "article": "Benign neoplasm of lymph nodes",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Drainage of Jejunum, Open Approach, Diagnostic",
      "image_url": "http://dummyimage.com/181x110.png/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Alcohol Prep Pads",
    },
    Object {
      "article": "Personal history of poisoning, presenting hazards to health",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement L Temporal Bone w Autol Sub, Perc Endo",
      "image_url": "http://dummyimage.com/130x171.bmp/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Alprazolam",
    },
    Object {
      "article": "Injury to other intra-abdominal organs with open wound into cavity, unspecified intra-abdominal organ",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement Upper Artery with Synth Sub, Perc Endo Approach",
      "image_url": "http://dummyimage.com/203x190.bmp/dddddd/000000",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Amoxicillin and Clavulanate Potassium",
    },
    Object {
      "article": "Open wound of nose, unspecified site, complicated",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Replace R Thumb Phalanx w Autol Sub, Perc Endo",
      "image_url": "http://dummyimage.com/145x196.jpg/5fa2dd/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Banana Boat Deep Tanning Dry SPF 8",
    },
    Object {
      "article": "Other noncollision motor vehicle traffic accident injuring unspecified person",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Removal of Int Fix from L Low Femur, Extern Approach",
      "image_url": "http://dummyimage.com/185x116.jpg/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Bethanechol Chloride",
    },
    Object {
      "article": "Open fracture of surgical neck of humerus",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Bypass Gastric Vein to Low Vein w Synth Sub, Open",
      "image_url": "http://dummyimage.com/186x136.jpg/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Bisoprolol Fumarate",
    },
    Object {
      "article": "Closed fracture of six ribs",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Replace of R Mandible with Nonaut Sub, Perc Endo Approach",
      "image_url": "http://dummyimage.com/183x196.bmp/5fa2dd/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "California Mugwort",
    },
    Object {
      "article": "Other degenerative disorders of globe",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Insertion of Ext Fix into L Humeral Head, Open Approach",
      "image_url": "http://dummyimage.com/113x143.jpg/cc0000/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "CataractClear",
    },
    Object {
      "article": "Amniotic fluid embolism, delivered, with mention of postpartum complication",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Restrict Esophagast Junct w Extralum Dev, Perc Endo",
      "image_url": "http://dummyimage.com/155x185.bmp/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Ciprofloxacin",
    },
    Object {
      "article": "Open wound of wrist, complicated",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Removal of Synthetic Substitute from Vas Deferens, Endo",
      "image_url": "http://dummyimage.com/146x212.jpg/dddddd/000000",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Coppertone ultraGUARD Sunscreen",
    },
    Object {
      "article": "Tuberculosis of intrathoracic lymph nodes, bacteriological or histological examination not done",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Dilate Mid Colic Art, Bifurc, w 3 Drug-elut, Open",
      "image_url": "http://dummyimage.com/139x102.jpg/cc0000/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Cover Fx Blemish Treatment Concealer N Deep",
    },
    Object {
      "article": "Open fracture of C1-C4 level with anterior cord syndrome",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Excision of Right Ventricle, Percutaneous Approach",
      "image_url": "http://dummyimage.com/130x134.png/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Cultivated Oat",
    },
    Object {
      "article": "Nephritis and nephropathy, not specified as acute or chronic, with lesion of rapidly progressive glomerulonephritis",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Removal of Autol Sub from Lum Disc, Perc Endo Approach",
      "image_url": "http://dummyimage.com/140x222.png/5fa2dd/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Cyanocobalamin",
    },
    Object {
      "article": "Lipidoses",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Restrict L Ext Iliac Vein w Intralum Dev, Perc",
      "image_url": "http://dummyimage.com/138x214.png/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "DYSPEPSIA HEADACHE",
    },
    Object {
      "article": "Nonspecific (abnormal) findings on radiological and other examination of abdominal area, including retroperitoneum",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Dilation of Right Hand Artery, Bifurcation, Open Approach",
      "image_url": "http://dummyimage.com/189x209.bmp/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Daysee",
    },
    Object {
      "article": "Infections of nipple associated with childbirth, postpartum condition or complication",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Occlusion R Ext Jugular Vein w Extralum Dev, Perc Endo",
      "image_url": "http://dummyimage.com/239x164.bmp/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Degree Girl Just Dance",
    },
    Object {
      "article": "Chronic total occlusion of coronary artery",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Removal of Synth Sub from C-thor Jt, Open Approach",
      "image_url": "http://dummyimage.com/201x168.jpg/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Doxazosin",
    },
    Object {
      "article": "Hypertrophy of tongue papillae",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Revision of Synth Sub in L Pelvic Bone, Extern Approach",
      "image_url": "http://dummyimage.com/193x103.jpg/5fa2dd/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Duet DHA",
    },
    Object {
      "article": "Other specified arthropod-borne viral diseases",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Replace of Nasal Septum with Synth Sub, Perc Endo Approach",
      "image_url": "http://dummyimage.com/103x250.jpg/cc0000/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Egg Yolk",
    },
    Object {
      "article": "Oligohydramnios, antepartum condition or complication",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement R Foot Muscle with Autol Sub, Perc Endo Approach",
      "image_url": "http://dummyimage.com/144x187.png/dddddd/000000",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Fresh Baby Scent Soothing Jelly",
    },
    Object {
      "article": "Illegally induced abortion, complicated by embolism, complete",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Change Other Device in Brain, External Approach",
      "image_url": "http://dummyimage.com/189x163.png/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Gianvi",
    },
    Object {
      "article": "Mixed hearing loss, unilateral",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Insert of Intralum Dev into R Innom Vein, Perc Endo Approach",
      "image_url": "http://dummyimage.com/189x221.bmp/dddddd/000000",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Healing",
    },
    Object {
      "article": "Pulp degeneration",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement Right Axilla with Synth Sub, Open Approach",
      "image_url": "http://dummyimage.com/240x111.bmp/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Health Mart Pharmacy Hydrocortisone",
    },
    Object {
      "article": "Mechanical loosening of prosthetic joint",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Detachment at Right Thumb, High, Open Approach",
      "image_url": "http://dummyimage.com/163x165.bmp/dddddd/000000",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Health Mart fexofenadine hydrochloride",
    },
    Object {
      "article": "Burkitt's tumor or lymphoma, spleen",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Release Right Upper Arm Tendon, Percutaneous Approach",
      "image_url": "http://dummyimage.com/117x193.bmp/cc0000/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Isosorbide Dinitrate",
    },
    Object {
      "article": "Wrist drop (acquired)",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Drainage of Cul-de-sac with Drainage Device, Open Approach",
      "image_url": "http://dummyimage.com/100x231.png/5fa2dd/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Kids Crest",
    },
    Object {
      "article": "Pulmonary alveolar proteinosis",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement R Low Leg Muscle w Nonaut Sub, Perc Endo",
      "image_url": "http://dummyimage.com/103x242.bmp/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "LOreal Paris Men Expert Vita Lift Daily Moisturizer Sunscreen Broad Spectrum SPF 15",
    },
    Object {
      "article": "Unspecified adverse effect of other drug, medicinal and biological substance",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Excision of Esophagus, Via Opening, Diagn",
      "image_url": "http://dummyimage.com/101x123.png/5fa2dd/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Losartan Potassium",
    },
    Object {
      "article": "Combinations of opioid type drug with any other drug dependence, episodic",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Excision of Lower Back, Open Approach, Diagnostic",
      "image_url": "http://dummyimage.com/230x181.png/ff4444/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Metoprolol Tartrate",
    },
    Object {
      "article": "Need for prophylactic vaccination and inoculation against tuberculosis [BCG]",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Change Packing Material on Left Hand",
      "image_url": "http://dummyimage.com/127x172.png/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Naltrexone Hydrochloride",
    },
    Object {
      "article": "Mixed hearing loss, unilateral with Apollo and Graphql",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Nextjs is good",
      "image_url": "http://dummyimage.com/189x221.bmp/dddddd/000000",
      "isPublished": true,
      "social": Array [
        "http://facebook.com",
        "http://twitter.com",
      ],
      "source": Array [
        "http://trustme.com",
        "http://someonetoldme.com",
      ],
      "tags": Array [
        "Apollo",
        "NextJs",
        "GraphQL",
      ],
      "title": "NextJs",
    },
    Object {
      "article": "\\"Light-for-dates\\"without mention of fetal malnutrition, 1,250- 1,499 grams",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Remove Tissue Expander from Up Extrem Subcu/Fascia, Perc",
      "image_url": "http://dummyimage.com/108x248.jpg/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Nighttime Sleep Aid",
    },
    Object {
      "article": "Spontaneous abortion, complicated by genital tract and pelvic infection, unspecified",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Replacement of R Com Iliac Art with Synth Sub, Open Approach",
      "image_url": "http://dummyimage.com/154x136.png/dddddd/000000",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "POA ANNUA POLLEN",
    },
    Object {
      "article": "Mechanical failure of instrument or apparatus during unspecified procedure",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Drainage of Right Choroid with Drain Dev, Perc Approach",
      "image_url": "http://dummyimage.com/248x211.bmp/dddddd/000000",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "PREVNAR 13",
    },
    Object {
      "article": "Nervous system complications from surgically implanted device",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Removal of Drain Dev from Tracheobronc Tree, Via Opening",
      "image_url": "http://dummyimage.com/139x181.png/cc0000/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "PURIFIED WATER",
    },
    Object {
      "article": "Open fracture of first cervical vertebra",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Revise Infusion Dev in R Metacarpocarp Jt, Extern",
      "image_url": "http://dummyimage.com/160x122.jpg/ff4444/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Phenobarbital",
    },
    Object {
      "article": "Cardiac catheterization as the cause of abnormal reaction of patient, or of later complication, without mention of misadventure at time of procedure",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Bypass L Atrium to R Pulm Art with Autol Vn, Open Approach",
      "image_url": "http://dummyimage.com/153x199.jpg/5fa2dd/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Proactiv",
    },
    Object {
      "article": "Tuberculosis of thyroid gland, unspecified",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Occlusion R Up Lobe Bronc w Intralum Dev, Perc Endo",
      "image_url": "http://dummyimage.com/153x215.jpg/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "SEROQUEL",
    },
    Object {
      "article": "Dermatitis due to unspecified substance taken internally",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Extirpation of Matter from Left Renal Artery, Open Approach",
      "image_url": "http://dummyimage.com/215x108.bmp/dddddd/000000",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "SHISEIDO THE SKINCARE TINTED MOISTURE PROTECTION",
    },
    Object {
      "article": "Spontaneous abortion, complicated by metabolic disorder, complete",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement L Glenoid Cav with Autol Sub, Perc Endo Approach",
      "image_url": "http://dummyimage.com/242x138.png/cc0000/ffffff",
      "isPublished": true,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "ShopRite Pain Relief PM",
    },
    Object {
      "article": "Tuberculosis of lung with cavitation, tubercle bacilli found (in sputum) by microscopy",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Fluoroscopy of Ileal Diversion Loop using L Osm Contrast",
      "image_url": "http://dummyimage.com/203x157.png/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "TERRACOTTA JOLI TEINT BEAUTIFYING FOUNDATION WITH SUNSCREEN SUN-KISSED, HEALTHY GLOW BROAD SPECTRUM SPF 20 NATURAL",
    },
    Object {
      "article": "Closed dislocation of finger, unspecified part",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Revision of Nonaut Sub in T-lum Jt, Perc Endo Approach",
      "image_url": "http://dummyimage.com/219x236.bmp/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "TOMMY GIRL ANTIPERSPIRANT",
    },
    Object {
      "article": "Peripheral T cell lymphoma, spleen",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Drainage of Left Lacrimal Bone, Perc Endo Approach, Diagn",
      "image_url": "http://dummyimage.com/188x190.png/cc0000/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Treatment Set TS350394",
    },
    Object {
      "article": "Abscess of salivary gland",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Supplement R Verteb Vein with Nonaut Sub, Open Approach",
      "image_url": "http://dummyimage.com/180x149.bmp/5fa2dd/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Trout",
    },
    Object {
      "article": "Benign neoplasm of rectum and anal canal",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Bypass R Axilla Art to L Extracran Art w Autol Art, Open",
      "image_url": "http://dummyimage.com/150x115.bmp/5fa2dd/ffffff",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "VUMON",
    },
    Object {
      "article": "Mild nonproliferative diabetic retinopathy",
      "author": Object {
        "email": "not allowed to see",
        "nickname": "bob",
      },
      "description": "Control Bleeding in Right Upper Arm, Perc Endo Approach",
      "image_url": "http://dummyimage.com/117x176.bmp/dddddd/000000",
      "isPublished": false,
      "social": Array [],
      "source": Array [],
      "tags": Array [],
      "title": "Venlafaxine Hydrochloride",
    },
  ],
  "pageInfo": Object {
    "endCursor": "Venlafaxine Hydrochloride",
    "hasNextPage": true,
  },
}
`;

exports[`Test pagination for articles Pagination Article 2`] = `
Object {
  "data": Object {
    "getArticles": Object {
      "edges": Array [
        Object {
          "article": "Toxoplasmosis of other specified sites",
          "author": Object {
            "email": "not allowed to see",
            "nickname": "bob",
          },
          "description": "Drainage of Upper Artery, Percutaneous Approach, Diagnostic",
          "image_url": "http://dummyimage.com/226x248.bmp/dddddd/000000",
          "isPublished": true,
          "social": Array [],
          "source": Array [],
          "tags": Array [],
          "title": "Vineyard Antibacterial Foaming Hand Wash",
        },
        Object {
          "article": "Injury to trigeminal nerve",
          "author": Object {
            "email": "not allowed to see",
            "nickname": "bob",
          },
          "description": "Dilation of L Axilla Art with Intralum Dev, Perc Approach",
          "image_url": "http://dummyimage.com/141x236.png/cc0000/ffffff",
          "isPublished": false,
          "social": Array [],
          "source": Array [],
          "tags": Array [],
          "title": "Zarontin",
        },
        Object {
          "article": "Infection of tracheostomy",
          "author": Object {
            "email": "not allowed to see",
            "nickname": "bob",
          },
          "description": "Release Right Kidney, Percutaneous Approach",
          "image_url": "http://dummyimage.com/218x134.jpg/dddddd/000000",
          "isPublished": false,
          "social": Array [],
          "source": Array [],
          "tags": Array [],
          "title": "ciprofloxacin",
        },
        Object {
          "article": "Deep necrosis of underlying tissues [deep third degree] with loss of a body part, of chin",
          "author": Object {
            "email": "not allowed to see",
            "nickname": "bob",
          },
          "description": "Dilate of R Ant Tib Art with 4+ Intralum Dev, Open Approach",
          "image_url": "http://dummyimage.com/131x242.png/dddddd/000000",
          "isPublished": true,
          "social": Array [],
          "source": Array [],
          "tags": Array [],
          "title": "equate daytime nitetime",
        },
        Object {
          "article": "Myogenic ptosis",
          "author": Object {
            "email": "not allowed to see",
            "nickname": "bob",
          },
          "description": "Extirpation of Matter from Nasal Septum, Perc Approach",
          "image_url": "http://dummyimage.com/234x168.jpg/cc0000/ffffff",
          "isPublished": true,
          "social": Array [],
          "source": Array [],
          "tags": Array [],
          "title": "good sense antacid",
        },
      ],
      "pageInfo": Object {
        "endCursor": "good sense antacid",
        "hasNextPage": false,
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
